components/logger.js

```diff
@@ -0,0 +1,107 @@
+import { createContext, useCallback, useContext, useEffect, useMemo, useState } from 'react'
+import { useMe } from './me'
+import fancyNames from '../lib/fancy-names.json'
+
+const generateFancyName = () => {
+  // 100 adjectives * 100 nouns * 10000 = 100M possible names
+  const pickRandom = (array) => array[Math.floor(Math.random() * array.length)]
+  const adj = pickRandom(fancyNames.adjectives)
+  const noun = pickRandom(fancyNames.nouns)
+  const id = Math.floor(Math.random() * fancyNames.maxSuffix)
+  return `${adj}-${noun}-${id}`
+}
+
+function detectOS () {
+  const userAgent = window.navigator.userAgent
+  const platform = window.navigator?.userAgentData?.platform || window.navigator.platform
```

Comment:
Here you check for navigator but you don't the line before. I think I recall navigator not being defined in some browsers. It might be best to bail out of os detection if navigator isn't defined.

Reply:
Copied that code [from SO](https://stackoverflow.com/a/38241481). Not sure why there is this inconsistency.

Also, `navigator` should be [supported by every browser](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/platform#browser_compatibility). We may had this error because of SSR?

But I'll use `window.navigator?` anyway. Better be safe than sorry :)